### ISSUE TITLE
The weight of a Excel file is bigger than pandas have created

### ISSUE THREAD

**Issue by IvanHod** [kz26/PyExcelerate#78]
When I created Excel files using **PyExcelerate** and **pandas**, the size of pandas xlsx file was less by 13mb. (48mb for PyExcelerate and 35mb for pandas). Can I decrease size of PyExcelerate file?
The table had 100 000 rows and 70 columns.

**Comment by kevmo314** [comment]
You can try compressing the xlsx file with an alternate compression algorithm. We use ZIP_DEFLATED but ZIP_LZMA might give you better results at a performance penalty. If you do find that it gives you better results maybe we can look into adding it as a tweakable argument to the writer.

**Comment by kevmo314** [comment]
You will need to modify the constant here: https://github.com/kz26/PyExcelerate/blob/247406dc41adc7e94542bcbf04589f1e5fdf8c51/pyexcelerate/Writer.py#L45

**Comment by IvanHod** [comment]
I have tested the compressing using an array of shape (50 000, 70)

Pandas converted the array to excel file for 87.97 seconds. Size of file is 44.7 mb. Next table has information for PyExcelerate about ZIP_DEFLATED and ZIP_BZIP2 modes, and different "compresslevel".
If to add compresslevel equals to 1, the size of file will decrease three times.

| compression - compresslevel | time, s | size of xlsx file, mb |
| --- | --- | --- |
| ZIP_DEFLATED-None | 48.891797 | 149.490331 |
| ZIP_DEFLATED-0 | 40.494316 | 149.490331 |
| ZIP_DEFLATED-1 | 43.405482 | 47.286776 |
| ZIP_DEFLATED-2 | 44.706557 | 46.443599 |
| ZIP_DEFLATED-3 | 45.615609 | 45.48636 |
| ZIP_DEFLATED-4 | 46.585664 | 44.836064 |
| ZIP_DEFLATED-5 | 48.611781 | 42.523898 |
| ZIP_DEFLATED-6 | 50.544891 | 42.372559 |
| ZIP_DEFLATED-7 | 52.223987 | 42.322049 |
| ZIP_DEFLATED-8 | 12.194129 | 42.098038 |
| ZIP_BZIP2-1 | 51.172927 | 31.469623 |
| ZIP_BZIP2-2 | 49.072807 | 31.678281 |
| ZIP_BZIP2-3 | 49.683841 | 31.766616 |
| ZIP_BZIP2-4 | 50.301878 | 32.093662 |
| ZIP_BZIP2-5 | 51.895968 | 32.118182 |
| ZIP_BZIP2-6 | 50.963915 | 32.203662 |
| ZIP_BZIP2-7 | 51.212929 | 32.270288 |
| ZIP_BZIP2-8 | 50.950914 | 32.338194 |

**Comment by kevmo314** [comment]
Yes this is partially an optimization. When we were implementing it, we found that building that table was nontrivially expensive for large sheets, so we opted to ignore it hoping that it would be a negligible cost due to a zip deflation. I see that's not the case though, I'll take another look at the shared strings table and see if we can optimize it behind an option or something.
